buffer/src/main/java/io/netty/buffer/UnpooledHeapByteBuf.java

```diff
@@ -536,9 +536,9 @@ protected void _setLongLE(int index, long value) {
     @Override
     public ByteBuf copy(int index, int length) {
         checkIndex(index, length);
-        byte[] copiedArray = PlatformDependent.allocateUninitializedArray(length);
-        System.arraycopy(array, index, copiedArray, 0, length);
-        return new UnpooledHeapByteBuf(alloc(), copiedArray, maxCapacity());
+        UnpooledHeapByteBuf copy = new UnpooledHeapByteBuf(alloc(), length, maxCapacity());
+        System.arraycopy(array, index, copy.array, 0, length);
```

Comment:
@njhill just to be 100 % safe should we better use: ? 

```
        System.arraycopy(array, index, copy.array, copy.arrayOffset(), length);
```

Reply:
or just:

```
alloc().heapBuffer(...)
...
...
```

Reply:
@normanmaurer good idea and that more closely aligns with the direct impl too, done.